metadata-ingestion/src/datahub/ingestion/glossary/datahub_classifier.py

```diff
@@ -81,28 +92,53 @@ class DataHubClassifierConfig(ConfigModel):
         description=f"List of infotypes to be predicted. By default, all supported infotypes are considered. If specified. this should be subset of {list(default_config.keys())}.",
     )
     info_types_config: Dict[str, InfoTypeConfig] = Field(
-        default={k: InfoTypeConfig.parse_obj(v) for k, v in default_config.items()},
+        default=DEFAULT_CLASSIFIER_CONFIG,
         init=False,
         description="Configuration details for infotypes. See [reference_input.py](https://github.com/acryldata/datahub-classify/blob/main/datahub-classify/src/datahub_classify/reference_input.py) for default configuration.",
     )
 
-    @root_validator
-    def provided_config_selectively_overrides_default_config(cls, values):
-        override: Dict[str, InfoTypeConfig] = values.get("info_types_config")
-        base = {k: InfoTypeConfig.parse_obj(v) for k, v in default_config.items()}
-        for k, v in base.items():
-            if k not in override.keys():
-                # use default InfoTypeConfig for info type key if not specified in recipe
-                values["info_types_config"][k] = v
+    @root_validator(pre=False)
+    def input_config_selectively_overrides_default_config(cls, values):
+        input_config: Dict[str, InfoTypeConfig] = values.get("info_types_config")
```

Comment:
It will be None if `info_types_config` doesn't exist, right?

Reply:
the data type of info_types_config is not Optional. also the default value is set, so I don't think this will ever be None. We already have tests that confirm this works fine without specifying info_types_config in config. Let me know if I am missing anything. I also moved this from root_validator to validator just now.